### PR TITLE
File type and Synthesized Filename parameters

### PR DESCRIPTION
**Pull request by mediumTaj**
### Summary

This pull request changes `file` type from `FileStream` to the more genreic `MemoryStream` so the user doesn't need to have data saved on disk. Since we now cannot get the filename from a `MemoryStream` I implemented synthesized filenames that some of the other SDKs use. Previously the SDK took byte[].

Additionally the SDK was regenerated using the latest OAS3 API definitions.

### CODE REVIEW

File: Scripts/Services/NaturalLanguageUnderstanding/V1/Model/AnalysisResults.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public class AnalysisResults
         /// API usage information for the request.
         /// </summary>
         [JsonProperty("usage", NullValueHandling = NullValueHandling.Ignore)]
-        public Usage Usage { get; set; }
+        public AnalysisResultsUsage Usage { get; set; }
```
Comment (mamoonraja) [inline Scripts/Services/NaturalLanguageUnderstanding/V1/Model/AnalysisResults.cs]: Why are we changing from `Usage` here? The new change means `AnalysisResults` has `AnalysisResultsUsage` in there. Using `AnalysisResults` again sounds repetitive. Is this change coming from `oas3`?

Comment (mediumTaj) [reply Scripts/Services/NaturalLanguageUnderstanding/V1/Model/AnalysisResults.cs]: Yes this is coming from the generator. I'm not sure how it synthesized the name. We can look into this if you think it's a big deal though

Comment (mediumTaj) [reply Scripts/Services/NaturalLanguageUnderstanding/V1/Model/AnalysisResults.cs]: Im trying to find where this gets transformed in the generator - Let me check with @padamstx tomorrow.

Comment (mamoonraja) [reply Scripts/Services/NaturalLanguageUnderstanding/V1/Model/AnalysisResults.cs]: We can resolve this conversation based on our talks offline!

**Pull request merged**